pkg/kubelet/cm/helpers.go

```diff
@@ -62,15 +63,26 @@ func buildContainerMapAndRunningSetFromRuntime(ctx context.Context, runtimeServi
 	runningSet := sets.New[string]()
 	containerMap := containermap.NewContainerMap()
 	containerList, _ := runtimeService.ListContainers(ctx, nil)
```

Comment:
if you want to filter out terminated containers, would it make sense to provide filter to the `ListContainers` instead of sorting them, adding to the map and then removing from it?

Reply:
Thanks, I will use filter in ListContainers

Reply:
While this is a good suggestion to compute more efficiently the `runningSet`, we need to ensure that doesn't change the `containerMap` content in an incompatible way

Reply:
The `containerMap` should contains all pod latest container, so i still list all container